oligo_designer_toolsuite/oligo_specificity_filter/_filter_blastn.py

```diff
@@ -187,15 +217,186 @@ def _find_hits(
+        bed = pd.concat([bed_plus, bed_minus])
+        # adjust for possible overflows (e.g. new coordinates are not included in the gene boundaries)
+        # additionally we store how muchpadding we have to do to have two seqeunces of the same length
+        bed["overflow_start"] = bed["start"].apply(lambda x: -x if x < 0 else 0)
+        bed["start"] = bed["start"].apply(lambda x: x if x >= 0 else 0)
+        # what about having the leght of a region as a method of the reference database?
+        records = SeqIO.index(file_reference, "fasta")
+        bed["len_region"] = bed["chr"].apply(lambda x: len(records[x].seq))
+        bed["overflow_end"] = bed[["end", "len_region"]].apply(
+            lambda x: x["end"] - x["len_region"] if x["end"] > x["len_region"] else 0,
+            axis=1,
+        )
+        bed["end"] = bed[["end", "len_region"]].apply(
+            lambda x: x["end"] if x["end"] <= x["len_region"] else x["len_region"],
+            axis=1,
+        )
+        bed.sort_index(inplace=True)
+        file_bed = os.path.join(self.dir_output, f"references_{region_id}.bed")
+        bed.to_csv(
+            file_bed,
+            sep="\t",
+            index=False,
+            header=False,
+            columns=["chr", "start", "end", "name", "score", "strand"],
+        )
+
+        # generate the fasta file
+        references_fasta_file = os.path.join(
+            self.dir_output, f"references_{region_id}.fasta"
+        )
+        get_sequence_from_annotation(
+            file_bed, file_reference, references_fasta_file, strand=True, nameOnly=True
+        )
+
+        references = [
+            off_reference.seq for off_reference in SeqIO.parse(references_fasta_file, "fasta")
+        ]
+        references_padded = []
+        for reference, overflow_start, overflow_end, strand in zip(
+            references, bed["overflow_start"], bed["overflow_end"], bed["strand"]
+        ):
+            if overflow_start != 0:
+                # add padding in front if on pls strand, at the end if on the minus strand
+                if strand == "+":
+                    reference = "-" * overflow_start + reference
+                elif strand == "-":
+                    reference = reference + "-" * overflow_start
+            if overflow_end != 0:
+                # add padding at the end if on the plus strand, at the beggining if on the minus strand
+                if strand == "+":
+                    reference = reference + "-" * overflow_end
+                if strand == "-":
+                    reference = "-" * overflow_end + reference
+            references_padded.append(reference)
+        os.remove(references_fasta_file)
+        os.remove(file_bed)
+        return references_padded
+
+    def _add_alignement_gaps(
+        self, table_hits: pd.DataFrame, queries: list, references: list
+    ):
+        """Adjust the sequences of the oligos to remove the gaps introduced by the alignement search.
+
```

Comment:
add or remove? the function is called add but in the doc string you say removed. It's a bit confusing